qemu/tests/cpu_flag_intel_pt.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+from avocado.utils import cpu, process
+from virttest import error_context, env_process
+from qemu.tests.x86_cpu_model import check_flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu flag intel-pt.
+    1) Check if current flags are in the supported lists, if no, cancel test
+    2) Otherwise, set pt_model = 1 on host at first
+    3) For q35
+       3.1) boot guest with cpu model 'host' without intel-pt
+       3.2) boot guest with cpu model 'Icelake-Server' with intel-pt
+    4) For pc
+       4.1) boot guest with intel-pt and min-level=0x14
+    5) Check cpu flags in guest(only for linux guest)
+    6) Reboot guest
+    7) Set pt_model = 0 on host at last
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def test_guest():
+        """
+        Check flags inside guest.
+        """
+        params["start_vm"] = "yes"
+        vm_name = params['main_vm']
+        env_process.preprocess_vm(test, params, env, vm_name)
+
+        vm = env.get_vm(vm_name)
+        error_context.context("Try to log into guest", logging.info)
+        session = vm.wait_for_login()
+        if params["os_type"] == "linux":
+            check_flags(params, flags, test, session)
+
+        if params.get("reboot_method"):
+            error_context.context("Reboot guest '%s'." % vm.name, logging.info)
+            session = vm.reboot(session=session)
+
+        vm.verify_kernel_crash()
+        session.close()
+        vm.destroy()
+
+    vendor_id = params.get("vendor_id", "")
+    if vendor_id:
+        if vendor_id != cpu.get_vendor():
+            test.cancel("Need host vendor %s to support this test case" % vendor_id)
+
+    flags = params["flags"]
+    check_host_flags = params.get_boolean("check_host_flags")
```

Comment:
You can't just check host flags for intel_pt. E.g. On Cascadelake host, host has intel_pt flag, but it doesn't support this feature.

Reply:
> You can't just check host flags for intel_pt. E.g. On Cascadelake host, host has intel_pt flag, but it doesn't support this feature.

How about we pick up the specific host to test the "intel-pt"? It is not flexible very much but very effective.

Reply:
Did you get any advice from developer?

Reply:
> Did you get any advice from developer?

not yet.